cmd/registry/cmd/upload/bulk/discovery.go

```diff
@@ -246,6 +206,18 @@ func (task *uploadDiscoveryTask) fetchDiscoveryDoc() error {
 	if err != nil {
 		return err
 	}
-	task.contents, err = core.GZippedBytes(bytes)
-	return err
+
+	// normalize the doc to produce consistent hashes
+	var m interface{}
+	err = json.Unmarshal(bytes, &m)
+	if err != nil {
+		return err
+	}
+	normalized, err := json.MarshalIndent(m, "", "  ")
+	if err != nil {
+		return err
+	}
```

Comment:
Consider moving this into a helper - the comment doesn't seem necessary if we name it `normalizeJSON`. It's not clear to me how this normalizes the JSON though... Even though we indent it consistently, wouldn't the fields still be out of order?

Reply:
"The json package always orders keys when marshalling." https://stackoverflow.com/a/18668885

Reply:
Right, so the use of MarshalIndent instead of Marshal is just for formatting. Thanks.

Reply:
Yes, the docs come indented from the Discovery service, so that seemed best.

I've added the helper that you suggested.

Reply:
Done.